contracts/src/v0.7/ErrorParser.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+/**
+ * @dev library for parsing hex-encoded errors external calls.
+ */
+library ErrorParser {
+
+  /**
+   * @notice extracts revert message from .call() result.
+   *
+   * @dev this extracts revert message by discarding the first 4 bytes of the
+   * encoded result, the signature for Error(msg), and decode the rest as string.
+   *
+   * @param result encoded bytes returned from .call()
+   * @return string decoded revert message.
+   */
+  function getRevertMessage(bytes memory result) internal pure returns (string memory) {
```

Comment:
Will this method work with Solidity v0.8's custom error types? I know they have a hash to identify them but I'm not sure if it is similarly truncated to 4 bytes at the start or not.

I know it improves readability to chop off the prefix, but I'm concerned that it may contain useful information. I'd advocate to leave the four bytes on there because the rest of the string can easily be decoded off-chain. I suspect long term the trend will move towards custom errors instead of old style revert strings.

Reply:
If that's the case, maybe having the logic in the methods is better instead of pulling it out as a library, this library was meant to hide that 4 bytes trimming from the contract but if we are keeping it, it should be ok. I'm not sure how abi.decode to string will handle the prefix as well but will play with it.

Reply:
as for the move to 0.8, doesn't that require moving all contracts forwarder depends on to 0.8/dev? 
OperatorInterface, ConfirmedOwnerWithProposal, AuthorizedReceiver, Address
if so, I'll start working into extracting the changes to 0.8 and move all the contracts it requires into dev

Reply:
Can you show me what a custom error looks like when returned from a contract call? Does it have a different 4 bytes to start with? A full 32 bytes, like a topic? I think that will make a difference of how we want to handle the first 4 bytes, maybe it is ok to cut it.

Reply:
I guess we would have to move over all of those contracts eventually, although not necessarily in this PR. Maybe it is easier for now to move this contract into `v0.7/dev` for now. We can figure out where this should move separately. I think moving it into dev for now should be ok since it is still pretty much unsued.

Reply:
I played with custom errors and it turns out the previous way I used for prefetching didn't work out well with that.Discussed with Yos on next steps we settled on propagating revert reason 1:1 with assembly. I updated the PR with that approach and cleaned up the extra library as the logic is much simpler now. and moving forward have prefixing will be the way to go for >=0.8
The previous method, gets rid of the method signature (usually Error but can have more context with custom errors i.e Unauthorized(), and also assumes one string parameter to that error ( Error(string) ) which doesn't work with custom errors

destination contract returning Unauthorized("username", false) will be propagated as `reverted with message: username` dropping signature and all other parameters.

Another approach is to manually construct the error in assembly but I believe deferring to more readable 0.8 stacktrace builder should be our way to go.